### PR TITLE
Added type hints at Europython sprint.

### LINKED ISSUE #9070: Improve bokeh/util/logconfig.py call signature
bokeh/util/logconfig.py 
in branch landing-2.0, version 1.1.0

regarding:
def basicConfig(*args, ''kwargs):

`logging.basicConfig(*args, **kwargs)`
but should only call 
`logging.basicConfig(**kwargs)`

*args should be removed from parameters, it seems that
no other code used args to call it..

### FOLLOW THESE INSTRUCTIONS CAREFULLY

*ISSUES THAT DO NOT CONTAIN NECESSARY INFORMATION MAY BE CLOSED, IMMEDIATELY*

The issue tracker is NOT the place for general support. For questions and technical assistance, come join the [Bokeh  Project Discourse](https://discourse.bokeh.org).

For defects or deficiencies, please provide *ALL OF THE FOLLOWING*:

#### Complete, minimal, self-contained example code that reproduces the issue

```
# code goes here between backticks

```

### PR DESCRIPTION
**Pull request by mborus**
Added type hints at Europython sprint.
also: Removed *args from logconfig.py basicConfig (Issue #9070)

All pull requests must have an associated issue in the issue tracker. If there
isn't one, please go open an issue describing the defect, deficiency or desired
feature. You can read more about our issue and PR processes in the
[wiki](https://github.com/bokeh/bokeh/wiki/BEP-1:-Issues-and-PRs-management).

- [ ] issues: fixes #9070
- [ ] tests added / passed
- [ ] release document entry (if new feature or API change)

### CODE REVIEW

**Comment by bryevdv** [comment]
here are the linter errors that failed the build:

 

>        assert proc.returncode == 0, "Flake8 issues:\n%s" % out.decode("utf-8")
> E       AssertionError: Flake8 issues:
> E         ./bokeh/util/terminal.py:24:1: F401 'typing.Tuple' imported but unused
> E         ./bokeh/util/terminal.py:24:1: F401 'typing.Dict' imported but unused
> E         ./bokeh/util/terminal.py:24:1: F401 'typing.Any' imported but unused
> E         ./bokeh/util/version.py:39:1: F401 'typing.Optional' imported but unused
> E         ./bokeh/util/version.py:39:1: F401 'typing.cast' imported but unused

File: bokeh/util/string.py [hunk 1]
```diff
@@ -1,33 +1,38 @@
...
+# -----------------------------------------------------------------------------
 ''' Functions useful for string manipulations or encoding.
 
 '''
 
-#-----------------------------------------------------------------------------
+# -----------------------------------------------------------------------------
 # Boilerplate
-#-----------------------------------------------------------------------------
+# -----------------------------------------------------------------------------
```
Comment (bryevdv) [inline bokeh/util/string.py]: I would actually ask that these spaces not be added. The convention in all other files is not to have a space before the `-` so changing here would introduce an inconsistency

Comment (mborus) [reply bokeh/util/string.py]: Spaces were autoadded by PyCharm, I have removed them, again.

Looking at the linter result now

File: bokeh/util/dependencies.py [hunk 2]
```diff
@@ -45,7 +46,7 @@
 # General API
 #-----------------------------------------------------------------------------
 
-def import_optional(mod_name):
+def import_optional(mod_name:str) -> Optional[ModuleType]:
```
Comment (mattpap) [inline bokeh/util/dependencies.py]: @mborus, please keep whitespace consistent here in a few other locations (one space after `:`).

File: bokeh/application/handlers/directory.py [hunk 3]
```diff
@@ -168,7 +166,7 @@ def safe_to_fork(self):
 
     # Public methods ----------------------------------------------------------
 
-    def modify_document(self, doc):
+    def modify_document(self, doc: Any) -> Any:
```
Comment (mattpap) [inline bokeh/application/handlers/directory.py]: A better signature would be `modify_document(self, doc: Document) -> None`, where `Document` comes from `bokeh.document`. The return type `None` acts like `void` in other languages and this method clearly doesn't return anything. The single `return` statement is just there to break execution early.

**Comment by mattpap** [comment]
Superseded by PR #9090.

**Pull request closed without merging**